test/livebook_teams/hubs/team_client_test.exs

```diff
@@ -548,6 +661,154 @@ defmodule Livebook.Hubs.TeamClientTest do
+      assert deployment_group in TeamClient.get_deployment_groups(team.id)
+
+      local = Livebook.FileSystem.Local.new()
+      path = Path.join(tmp_dir, "MyNotebook2.livemd")
+      file = Livebook.FileSystem.File.new(local, path)
+      slug = Livebook.Utils.random_short_id()
+
+      notebook = %{
+        Livebook.Notebook.new()
+        | app_settings: %{Livebook.Notebook.AppSettings.new() | slug: slug},
+          name: "MyNotebook2",
+          hub_id: team.id,
+          deployment_group_id: to_string(deployment_group.id)
+      }
+
+      {notebook_content, []} = Livebook.LiveMarkdown.notebook_to_livemd(notebook)
+      :ok = Livebook.FileSystem.File.write(file, notebook_content)
+
+      data = %Livebook.Session.Data{file: file, notebook: notebook}
+
+      {:ok, app_deployment} = Livebook.Teams.AppDeployment.new(data)
+      :ok = Livebook.FileSystem.File.remove(file)
+      {:ok, zip_content} = Livebook.FileSystem.File.read(app_deployment.file)
+
+      secret_key = Livebook.Teams.derive_key(team.teams_key)
+      encrypted_content = Livebook.Teams.encrypt(zip_content, secret_key)
+
+      bypass = Bypass.open()
+
+      Bypass.expect_once(bypass, "GET", "/#{app_deployment.filename}", fn conn ->
+        conn
+        |> Plug.Conn.put_resp_content_type("text/plain")
+        |> Plug.Conn.resp(200, encrypted_content)
+      end)
+
+      # Since the app deployment struct generation is from Livebook side,
+      # we don't have yet the information about who deployed the app,
+      # so we need to add it ourselves
+      app_deployment = %{
+        app_deployment
+        | id: "1",
+          deployed_by: "Jake Peralta",
+          deployed_at: NaiveDateTime.utc_now()
+      }
+
+      # updates the deployment group with an app deployment
+      # and after we deploy, the `:file` key turns to `nil` value
+      updated_deployment_group = %{
+        deployment_group
+        | app_deployments: [%{app_deployment | file: nil}]
+      }
+
+      livebook_proto_app_deployment =
+        %LivebookProto.DeployedApp{
+          id: app_deployment.id,
+          title: app_deployment.title,
+          slug: app_deployment.slug,
+          sha: app_deployment.sha,
+          archive_url: "http://localhost:#{bypass.port}/#{app_deployment.filename}",
+          deployed_by: app_deployment.deployed_by,
+          deployed_at: to_string(app_deployment.deployed_at),
+          app_id: "1",
+          deployment_group_id: app_deployment.deployment_group_id
+        }
+
+      updated_livebook_proto_deployment_group = %{
+        livebook_proto_deployment_group
+        | deployed_apps: [livebook_proto_app_deployment]
+      }
+
+      agent_connected = %{
+        agent_connected
+        | deployment_groups: [updated_livebook_proto_deployment_group]
+      }
+
+      apps_path = Path.join(tmp_dir, "apps")
+      app_path = Path.join(apps_path, slug)
+      Application.put_env(:livebook, :apps_path, apps_path)
```

Comment:
We want to avoid `Application.put_env` in async tests to avoid race conditions. But we shouldn't need to change the apps path anyway, since `Apps.deploy_apps_in_dir` accepts the dir as an argument.

Reply:
I can remove this if we start using the `tmp_dir` to deploy the applications from the Teams API